lib/pbench/test/functional/server/test_put.py

```diff
@@ -21,38 +21,46 @@ class Tarball:
 
 
 class TestPut:
-    tarballs: dict[str, Tarball] = {}
+    """These tests depend on the order of their definition to execute properly.
 
-    def test_upload_all(self, server_client: PbenchServerClient, login_user):
+    That is, test_index_all assumes that all the tar balls in the `TARBALL_DIR`
+    were uploaded successfully, and that it is run before `test_delete`. In
+    turn, `test_delete` must be run before `test_upload_all`.
+    """
+
+    @staticmethod
+    def test_upload_all(server_client: PbenchServerClient, login_user):
```

Comment:
I can see why you might want to run `test_delete` _before_ running `test_upload_all` (to establish a clean slate on the Server), but since `test_upload_all` seems to appear first in the class, I suspect that it gets run before `test_delete` (which also makes good sense).

So, I think the comment text needs tweaking.

Reply:
Running `test_delete` first would make the test module idempotent, which might be convenient in the iterative "live server" case, but normally we definitely want to delete when we're done or it's not testing anything. I don't think we have any automatic way of running it twice; so unless you want to duplicate it with different names I think it's reasonably incumbent on the "live server iterative tester" with a failure case to manually run `test_delete` before starting a subsequent cycle...